resources/hpxml-measures/workflow/tests/compare.py

```diff
@@ -57,7 +56,51 @@ def results(self, base_folder, feature_folder, export_folder, groupby=[]):
         df = df.astype(int)
 
       df = df.fillna('NA')
-      df.to_csv(os.path.join(export_folder, file))
+      df.to_csv(os.path.join(export_folder, file))    
+
+      # Get results charactersistics of groupby columns
+      if file == 'results_characteristics.csv':
+        if 'build_existing_model.geometry_building_type_recs' not in groupby:
+          groupby.append('build_existing_model.geometry_building_type_recs')
+        group_df = base_df[groupby]
+
+      # Write grouped & aggregated results dfs
+      if file == 'results_output.csv':
+        # Map building types
+        if 'build_existing_model.geometry_building_type_recs' in groupby:
+          btype_map = {'Single-Family Detached': 'SFD',
+                      'Single-Family Attached': 'SFA',
+                      'Multi-Family with 2 - 4 Units': 'MF',
+                      'Multi-Family with 5+ Units': 'MF'}
+          group_df['build_existing_model.geometry_building_type_recs'] = group_df['build_existing_model.geometry_building_type_recs'].map(btype_map)
+
+        # Merge groupby df and aggregate
+        base_df = group_df.merge(base_df, 'outer', left_index=True, right_index=True).groupby(groupby)
+        base_df = base_df.sum().stack()
+        feature_df = group_df.merge(feature_df, 'outer', left_index=True, right_index=True).groupby(groupby).sum()
+        feature_df = feature_df.stack()
+
+    # Write aggregate results df
+    deltas = pd.DataFrame()
+    deltas['base'] = base_df
+    deltas['feature'] = feature_df
+    deltas['diff'] = deltas['feature'] - deltas['base']
+    deltas['% diff'] = 100*(deltas['diff']/deltas['base'])
+    deltas = deltas.round(2)                          
+    deltas.reset_index(level=groupby, inplace=True)
+    deltas.index.name = 'enduse'
+    sims_df = pd.DataFrame({'base':len(base_df),
+                            'feature':len(feature_df),
+                            'diff':'n/a',
+                            '% diff':'n/a'},
+                            index=['simulation_count'])
+    sims_df[groupby] = 'n/a'
+    deltas = pd.concat([sims_df, deltas])
+    for group in groupby:
+      first_col = deltas.pop(group)
+      deltas.insert(0, group, first_col)
+
+    deltas.to_csv(os.path.join(export_folder, 'aggregate_results.csv'))
```

Comment:
@aspeake Can you update this filename to be the original filename, then underscore, then the column you are grouping by? (See below where this is happening for plot filenames.)

Reply:
@joseph-robertson yes, we still need to write it with some new prefix (ie `aggregate_results_*`) to differentiate from the results_output.csv though, right?

Reply:
@aspeake I don't think so. These results are exported to a separate folder (i.e., `comparisons` instead of `base`).

Reply:
@joseph-robertson But it looks like there is already a `comparisons/results_output.csv` file that gets generated, which compares buildings 1:1